pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -192,14 +195,26 @@ func (lb *ListenerBuilder) buildInboundListeners() []*listener.Listener {
 
 // inboundVirtualListener builds the virtual inbound listener.
 func (lb *ListenerBuilder) inboundVirtualListener(chains []*listener.FilterChain) *listener.Listener {
-	actualWildcard, _ := getActualWildcardAndLocalHost(lb.node)
+	var actualWildcard, extrActualWildcard string
+	actualWildcard, extrActualWildcard = getDualStackActualWildcard(lb.node)
+	if actualWildcard == "" && extrActualWildcard == "" {
+		actualWildcard, _ = getActualWildcardAndLocalHost(lb.node)
+	}
 
 	// Build the "virtual" inbound listener. This will capture all inbound redirected traffic and contains:
 	// * Passthrough filter chains, matching all unmatched traffic. There are a few of these to handle all cases
 	// * Service filter chains. These will either be for each Port exposed by a Service OR Sidecar.Ingress configuration.
 	allChains := buildInboundPassthroughChains(lb)
 	allChains = append(allChains, chains...)
-	return lb.buildInboundListener(model.VirtualInboundListenerName, util.BuildAddress(actualWildcard, ProxyInboundListenPort), false, allChains)
+	l := lb.buildInboundListener(model.VirtualInboundListenerName, util.BuildAddress(actualWildcard, ProxyInboundListenPort), false, allChains)
```

Comment:
should we do this in buildInboundListener?

Reply:
Hi @howardjohn, I think it's better to separate the default service and extra service address, because 1. no change for `buildInboundListener` ； 2. no impact for single stack logic; 3.  Extra service address should be added separately, because it's not just for dual stack, it may be used for other in future.   Any idea?